packages/@aws-cdk/aws-codepipeline/lib/stage.ts

```diff
@@ -147,6 +147,10 @@ export class Stage extends cdk.Construct implements actions.IStage, actions.IInt
     return this.pipeline.role;
   }
 
+  public get pipelineAsEventTarget(): events.IEventRuleTarget {
```

Comment:
Why is this needed? The caller can just do `stage.pipeline` - it will also be much more readable at the call site.

Reply:
A caller cannot do `stage.pipeline`, because you cannot reference the `codepipeline` module from the `codepipeline-api` one (where the `IStage` interface lives).

Reply:
Got it